metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -392,6 +425,17 @@ final class ReferenceProvider(
     buf.result().sortWith(sortByLocationPosition)
   }
 
+  private def getOptImportRenameLocation(
+      range: s.Range,
+      source: AbsolutePath,
+  ): Option[s.Range] = {
+    for {
+      rename <- trees
```

Comment:
The file might not be open, so the Trees will not contain the parsed tree. I think we can skip references in that case.

Alternatively, we could try to manually find `=>` or `as` and the identifier after it.

Reply:
The problem is we don't know if we even are in an import statement, so looking for `=>/as` would give many false positives. Can I just parse the file? If I can cache the result it shouldn't be a waste. I can use the `=>/as` as a precondition, to limit the number of files I have to parse. (I was looking at other possibilities like regexes to validate the surroundings of the occurrence, but that's error prone and I'm not convinced it will be more performant than parsing the file).

Reply:
Let's just skip it, so let's show the references when we are on a rename, but we can skip it the other way around. I think it would be too much effort and the gain is minimal. It will show the symbol pre `=>` so that's ok